test/blackbox-tests/test-cases/custom-cross-compilation/toolchain.t

```diff
@@ -0,0 +1,82 @@
+Test the toolchain field in the workspace.
+
+First, we create the project
+  $ cat >dune <<EOF
+  > (library
+  >  (public_name foo))
+  > EOF
+
+  $ cat >dune-project <<EOF
+  > (lang dune 3.10)
+  > (package
+  >  (name foo))
+  > EOF
+
+  $ buildfoo() {
+  >   dune clean
+  >   dune build @install -x foobar
+  >   [ -e _build/default.foobar/foo.cma ] && echo built foo.cma
+  > }
+
+Now the toolchain:
+
+Now we create the toolchain and make sure we actually use it:
+
+  $ actualocamlc="$(command -v ocamlc)"
+
+  $ mkdir -p etc/findlib.conf.d
+  $ cat >etc/findlib.conf <<EOF
+  > path=""
+  > ocamlc="$PWD/notocamlc"
+  > EOF
+  $ cat >etc/findlib.conf.d/foo.conf <<EOF
+  > path(foobar)=""
+  > ocamlc(foobar)="$PWD/notocamlc-foobar"
+  > EOF
+
+  $ cat >notocamlc <<EOF
+  > #!/usr/bin/env sh
+  > echo "from notocamlc" >&2
+  > $actualocamlc \$@
+  > EOF
+  $ cat >notocamlc-foobar <<EOF
+  > #!/usr/bin/env sh
+  > echo "from notocamlc-foobar" >&2
+  > $actualocamlc \$@
+  > EOF
+  $ chmod +x notocamlc notocamlc-foobar
+
+We set an undefined toolchain with findlib:
+
+  $ OCAMLFIND_TOOLCHAIN=foobar buildfoo
```

Comment:
was confused because the PR description said `OCAML_TOOLCHAIN`, but I see that was just a typo and this is correct.